arcsi/api/show.py

```diff
@@ -389,30 +389,52 @@ def view_show_archive(show_slug):
 @auth_token_required
 def view_show_page(show_slug):
     do = DoArchive()
+    archived = request.args.get('archived', True, type=lambda v: v.lower() == 'true')
```

Comment:
Thanks for implementing the idea of these parameter controlled dynamic views. For a couple of reasons, which I will try to convince you about please consider to change almost the whole logic of this part of your changes, or pull these out of this PR, and have a separate branch where we can work on only this as a focus.

First, these two variables are used for a clear goal here, to decide what data to send back with the view. That's a good intent! The way it is defined is bothering me. I will try to explain what parts need to be thought about and will try to include how I would go about and what we benefit.

0, `Archived` and `latest` might be useful in the Show api in other parts of the Api. I would place these variables out from method scope. Then all public endpoints are aware of the same arguments and state.
1, There should be one method in the module that does what these lines. I'm looking at the `utils` pack and imagine I would add there a short function that;
 - imports the request context, 
 - takes a string we want as input and 
 - lists all argument keys searching for it. 
(It can return a boolean or the key or the value, it isn't that important.) I would look into `after_request` so we don't have to call parameter parsing on each endpoint and that has benefits like simplicity in the HTTP handling part, or that we can apply these for all views in general
2, Here we are returning the Boolean. It is a weird way to do I would definitely think of just having `?archived` and handling that (this `lambda` implies to me that we handle `?archived=True` is that correct or not?) 

Now don't take me wrong, I believe your are exercising the good practice of gardening where you always improve things in small bits as you find them that I appreciate a lot. I would like to only highlight that internal improvements of the code have to be a top consideration also.

Reply:
After we discussed this part I can agree with you, I added a little workaround to replace the previously used default values (`archived=True` and `latest=False`, they are needed because the frontend already using this API without any params)